components/formats-bsd/src/loci/formats/tiff/TiffSaver.java

```diff
@@ -1021,13 +1021,22 @@ private void makeValidIFD(IFD ifd, int pixelType, int nChannels) {
     if (nChannels == 1 && ifd.getIFDValue(IFD.COLOR_MAP) != null) {
       pi = PhotoInterp.RGB_PALETTE;
     }
-    else if (nChannels == 3) {
+    else if (nChannels == 3 || nChannels == 4) {
+      // TODO: handle CMYK; nChannels == 4, extraSamples 0 (can be omitted)
       if (ifd.getIFDValue(IFD.COMPRESSION).equals(TiffCompression.JPEG.getCode())) {
         // see https://github.com/ome/bioformats/issues/3856
         pi = PhotoInterp.Y_CB_CR;
+        if (nChannels == 4) {
+          int[] extraSamplesArray = {2};
```

Comment:
As in line 1037, I am unconvinced that we can safely assume that the 4th channel is "unassociated alpha data". I would prefer to see this value changed to 0 unless there is a strong argument to the contrary.